sky/serve/autoscalers.py

```diff
@@ -318,12 +333,45 @@ def get_latest_version_with_min_replicas(
     def select_outdated_replicas_to_scale_down(
             self, replica_infos: Iterable['replica_managers.ReplicaInfo']
     ) -> List[int]:
+        """Select outdated replicas to scale down."""
+
+        latest_ready_replicas = []
+        old_replicas = []
+        for info in replica_infos:
+            if info.version == self.latest_version:
+                if info.is_ready:
+                    latest_ready_replicas.append(info)
+            else:
+                old_replicas.append(info)
+
+        num_latest_ready_replicas = len(latest_ready_replicas)
+        if self.update_mode == serve_utils.UpdateMode.ROLLING:
+            if num_latest_ready_replicas >= self.target_num_replicas:
```

Comment:
For rolling, we use `self.target_num_replica`, for blue green, we are using `min_replicas`. Maybe add an explanation here.